velox/connectors/hive/HiveDataSink.h

```diff
@@ -205,10 +214,21 @@ class HiveDataSink : public DataSink {
   const std::shared_ptr<const HiveInsertTableHandle> insertTableHandle_;
   const ConnectorQueryCtx* connectorQueryCtx_;
   const CommitStrategy commitStrategy_;
-  // Parameters used by writers, and thus are tracked in the same order
-  // as the writers_ vector
+  const std::vector<column_index_t> partitionChannels_;
+  const std::shared_ptr<PartitionIdGenerator> partitionIdGenerator_;
+
+  // Below are structures for partitions from all inputs. writerInfo_ and
+  // writers_ are both indexed by partitionId.
   std::vector<std::shared_ptr<HiveWriterInfo>> writerInfo_;
   std::vector<std::unique_ptr<dwrf::Writer>> writers_;
+
+  // Below are structures updated when generating partition IDs for the current
+  // input. input_ and partitionIds_ are indexed by the row of input_.
+  // partitionRows_ and partitionSizes_ are indexed by partitionId.
+  RowVectorPtr input_;
```

Comment:
This member variable should not be needed.

Reply:
Removed